### PR TITLE
implement primary-selection-unstable-v1 protocol

### PR DESCRIPTION
**Pull request by e2dk4r**
this commit replaces `gtk-primary-selection` protocol with `primary-selection-unstable-v1` for more cross compositor support.

| protocol                              | Mutter | KWin | Sway | Weston | Mir |
|---------------------------------------| ------ | ---- | ---- | ------ | --- |
|zwp_primary_selection_device_manager_v1|      1 |    1 |    1 |  x     | 1   |

see: https://wayland.app/protocols/primary-selection-unstable-v1#compositor-support

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -36,6 +40,13 @@ $(OBJ): $(GEN)
 %.h: %.xml
 	$(WAYLAND_SCANNER) client-header < $< > $@
 
+primary-selection-unstable-v1.xml:
+	cp $(WAYLAND_PROTOCOLS_DIR)/unstable/primary-selection/primary-selection-unstable-v1.xml ./
+xdg-decoration-unstable-v1.xml:
+	cp $(WAYLAND_PROTOCOLS_DIR)/unstable/xdg-decoration/xdg-decoration-unstable-v1.xml ./
+xdg-shell.xml:
+	cp $(WAYLAND_PROTOCOLS_DIR)/stable/xdg-shell/xdg-shell.xml ./
+
```
Comment (ii8) [inline Makefile:49]: If they are copied here like this they need to be removed also by the clean target. What's the benefit of doing it this way?

File: Makefile [hunk 2]
```diff
@@ -9,14 +9,18 @@ VERSION="0.5.0"
...
-GEN=xdg-shell.c xdg-shell.h xdg-decoration-unstable-v1.c \
-	xdg-decoration-unstable-v1.h gtk-primary-selection.c gtk-primary-selection.h
+OBJ=xdg-shell.o \
+		xdg-decoration-unstable-v1.o \
+		primary-selection-unstable-v1.o \
+		glyph.o \
+		main.o
+GEN=xdg-shell.c xdg-shell.h \
+		xdg-decoration-unstable-v1.c xdg-decoration-unstable-v1.h \
+		primary-selection-unstable-v1.h primary-selection-unstable-v1.c
```
Comment (ii8) [inline Makefile:20]: Nitpicks but: It should be lined up at tabstop 8, not 2. And there's a trailing space.

**Review by ii8 — CHANGES_REQUESTED** [review]
Thanks, I've been meaning to do this. I just have a few comments.

File: Makefile [hunk 3]
```diff
@@ -9,14 +9,18 @@ VERSION="0.5.0"
...
+		primary-selection-unstable-v1.o \
+		glyph.o \
+		main.o
+GEN=xdg-shell.c xdg-shell.h \
+		xdg-decoration-unstable-v1.c xdg-decoration-unstable-v1.h \
+		primary-selection-unstable-v1.h primary-selection-unstable-v1.c 
 
 havoc: tsm $(OBJ)
-	$(CC) $(LDFLAGS) -o $@ $(OBJ) $(LIBS)
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(OBJ) $(LIBS)
```
[the remark this answers is not in the document]
Comment (e2dk4r) [reply Makefile:23]: My bad. I was testing things

File: Makefile (hunk 1 above)
Comment (e2dk4r) [reply Makefile:49]: When I cloned the repo, `make` failed due to protocol files not found. this makes sure that it is copied from `/usr/share/wayland-protocols` to current directory.
This way makes cross distro (ex: debian, arch) builds more easy for package maintainers.
I might be wrong.

**Comment by e2dk4r** [comment]
This is closed because, I cloned shallow copy and force pushed it as mistake.

File: Makefile (hunk 1 above)
Comment (ii8) [reply Makefile:49]: Hm that's no good, why did make not find the files? VPATH should tell make to go look in those directories for source files. Is this way maybe portable to non-gnu make, are you using a bsd make or something?
Could you share the whole error output when it fails?
Why do you think it makes a difference for cross distro builds? Don't both methods require the files to be in the same place.

**Pull request closed without merging**